src/ring_theory/hahn_series.lean

```diff
@@ -797,14 +797,19 @@ variables {Γ' : Type*} [ordered_cancel_add_comm_monoid Γ']
         (function.embedding.prod_map ⟨f, hfi⟩ ⟨f, hfi⟩),
         (emb_domain ⟨⟨f, hfi⟩, hf⟩ x).coeff (ij.1) *
         (emb_domain ⟨⟨f, hfi⟩, hf⟩ y).coeff (ij.2),
-      { simp },
+      { simp only [rel_embedding.coe_fn_mk, hahn_series.emb_domain_mk_coeff, hahn_series.mul_coeff,
+          _root_.prod_map, finset.sum_map, finset.sum_congr,
+          function.embedding.coe_fn_mk, function.embedding.coe_prod_map], },
```

Comment:
@awainverse, this proof started timing out. I'm not exactly sure why, but I have "fixed" it by replacing some terminal simps with simp only.

Could you confirm you're okay with this, and/or golf or speed-up this proof, so it's no longer near the timeout limit? Is there a subsidiary lemma you could pull out?

Reply:
Those changes seem fine. I will also look to see if I can speed this up.